### PR TITLE
udev: net_id - support predictable ifnames on vio buses

### PR DESCRIPTION
**Pull request by fbuihuu**
For IBM PowerVM Virtual I/O network devices, we can build predictable names
based on the slot number passed as part of the OF "reg" property.  Valid slot
numbers range between 2-32767, so we only need the bottom half of the unit
address passed.

For example:

  /proc/device-tree/vdevice/l-lan@30000002
  /proc/device-tree/vdevice/vnic@30000005

would initially map to something like:

  /sys/devices/vio/30000002/net/eth0
  /sys/devices/vio/30000005/net/eth1

and would then translate to env2 and env5

This patch ignores the bus number, as there should only ever be one bus, and
then remove leading zeros.

### CODE REVIEW

File: src/udev/udev-builtin-net_id.c [hunk 1]
```diff
@@ -45,6 +45,7 @@
  *                                         — PCI geographical location
  *   [P<domain>]p<bus>s<slot>[f<function>][u<port>][..][c<config>][i<interface>]
  *                                         — USB port number chain
+ *   v<slot>                               - VIO slot number
```
Comment (poettering) [inline src/udev/udev-builtin-net_id.c]: can you maybe add a reference to "IBM PowerVM" or so to this comment? I for one wouldn't know what VIO is...

**Comment by poettering** [comment]
looks superficially OK to me. Of course I don't have any such IBM hw, hence I'd just merge this without really grokking whether this makes sense.

If you can get somebody from the IBM community to review this too, this would be great.

**Comment by fbuihuu** [comment]
On 03/30/2017 06:35 PM, Lennart Poettering wrote:
> If you can get somebody from the IBM community to review this too, this
> would be great.
> 

I think it's possible. This code has already been tested by someone from
IBM andI I guess he will be fine to give his ACK here.

**Comment by poettering** [comment]
> I think it's possible. This code has already been tested by someone from IBM andI I guess he will be fine to give his ACK here.

That'd be excellent (though I'd merge it even without...)

**Comment by fbuihuu** [comment]
On 03/30/2017 06:34 PM, Lennart Poettering wrote:
> 
> can you maybe add a reference to "IBM PowerVM" or so to this comment? I
> for one wouldn't know what VIO is...
> 

Ok I will, I must admit that I didn't know about VIO until a couple days ;)

File: src/udev/udev-builtin-net_id.c (hunk 1 above)
Comment (solgato) [reply src/udev/udev-builtin-net_id.c]: A good summary/overview of IBM PowerVM is the following Redbook
[](url)http://www.redbooks.ibm.com/abstracts/sg247940.html

Chapter 17 of the following LoP draft document contains deeper details on the Virtual I/O layer and handling:
[](url)https://openpowerfoundation.org/wp-content/uploads/2016/05/LoPAPR_DRAFT_v11_24March2016_cmt1.pdf

**Comment by poettering** [comment]
lgtm

**Comment by poettering** [comment]
```
KVM: entry failed, hardware error 0x0
EAX=00000000 EBX=00000000 ECX=00000000 EDX=000206a1
ESI=00000000 EDI=00000000 EBP=00000000 ESP=00000000
EIP=0000fff0 EFL=00000002 [-------] CPL=0 II=0 A20=1 SMM=0 HLT=0
ES =0000 00000000 0000ffff 00009300
CS =f000 ffff0000 0000ffff 00009b00
SS =0000 00000000 0000ffff 00009300
DS =0000 00000000 0000ffff 00009300
FS =0000 00000000 0000ffff 00009300
GS =0000 00000000 0000ffff 00009300
LDT=0000 00000000 0000ffff 00008200
TR =0000 00000000 0000ffff 00008b00
GDT=     00000000 0000ffff
IDT=     00000000 0000ffff
CR0=60000010 CR2=00000000 CR3=00000000 CR4=00000000
DR0=0000000000000000 DR1=0000000000000000 DR2=0000000000000000 DR3=0000000000000000 
DR6=00000000ffff0ff0 DR7=0000000000000400
EFER=0000000000000000
Code=ff ff 66 5b 66 83 c4 08 66 5b 66 5e 66 c3 b0 20 e6 20 66 c3 <ea> 5b e0 00 f0 30 36 2f 32 33 2f 39 39 00 fc 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
```
CI failure appears unrelated, merging.

**Pull request merged**